app_dart/lib/src/foundation/utils.dart

```diff
@@ -103,6 +103,22 @@ Future<List<Map<String, dynamic>>> getRepoBuilders(HttpClientProvider branchHttp
       .toList();
 }
 
+/// Returns [builder] when any file falls in any pre-set dir.
+Future<List<Map<String, dynamic>>> getFilteredBuilders(List<Map<String, dynamic>> builders, List<String> files) async {
+  final List<Map<String, dynamic>> filteredBuilders = <Map<String, dynamic>>[];
+  for (Map<String, dynamic> builder in builders) {
+    final List<String> dirs = List<String>.from((builder['run_if'] as List<dynamic>) ?? <String>['']);
+    for (String dir in dirs) {
```

Comment:
How do we support * and **?

Reply:
Existing cirrus seems to use **  only, do we have the use case of a single *?

Reply:
add * support.

Reply:
should we call it glob instead of dirs? dirs is confusing as it can be a file a folder or a glob, glob matches all of them.

Reply:
Replaced.